modules/system-disaster-recovery/src/integrationTest/java/org/apache/ignite/internal/disaster/system/ItCmgDisasterRecoveryTest.java

```diff
@@ -111,6 +128,11 @@ private CompletableFuture<Void> restartOrShutdownFuture(int nodeIndex) {
         return ((IgniteServerImpl) cluster.server(nodeIndex)).restartOrShutdownFuture();
     }
 
+    private static ClusterState clusterState(IgniteImpl restartedIgniteImpl1)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        return restartedIgniteImpl1.clusterManagementGroupManager().clusterState().get(10, SECONDS);
```

Comment:
Why not the classic `assertThat(fut, willCompleteSuccessfully())` and then `fut.join()`?

Reply:
What you suggest requires 3 lines:
```
CompletableFuture<T> future = blablaFuture();
assertThat(future, willCompleteSuccessfully());
return future.join();
```
But with timed get we only need one line:
```
return blablaFuture().get(...);
```
The matcher-based approach is good when you don't need the result of the future, but here produces a code that is too cumbersome.

Reply:
We use this construction quite often, I agree that it requires more lines of code. I don't insist, it's up to you.
Example: `org.apache.ignite.internal.cluster.management.ItClusterManagerTest#findLeaderNode`

Reply:
`.get(...)` declares checked exceptions. In `findLeaderNode()` we would have to handle them with a try/catch, bloating the 'just one line' to 'many lines'. But I used this in contexts where we can just declare those `throws` ourselves, so the bloating does not happen.